rasa/shared/core/slot_mappings.py

```diff
@@ -167,6 +167,39 @@ def entity_is_desired(
 
         return slot_fulfils_entity_mapping
 
+    @staticmethod
+    def check_mapping_validity(mapping: Dict[Text, Any], domain: "Domain") -> bool:
```

Comment:
Let's also add the `slot.name` as argument, and then that can be given in the warning message to clarify which slot is the problematic slot.